src/puppetlabs/puppetdb/command.clj

```diff
@@ -302,11 +303,45 @@
 (def supported-command?
   (comp (kitchensink/valset command-names) :command))
 
+(defprotocol PuppetDBCommandDispatcher
+  (enqueue-command [this connection endpoint command version payload])
+  (enqueue-raw-command [this connection endpoint raw-command])
+  (stats [this]))
```

Comment:
Would like to see a docstring or something talking about what `stats` returns, and about the difference between `enqueue-command` and `enqueue-raw-command`.